### PR TITLE
irc: add support for CAP cap-notify

### PR DESCRIPTION
**Pull request by maxteufel**

### CODE REVIEW

**Comment by Mikaela** [comment]
Is there something wrong with PR or when can we expect this to come to WeeChat?

If I have cap-notify enabled and away-notify in irc.server.blah.capabilities and blah starts offering away-notify, will it be automatically enabled? There was some script which required everything to be configured manually and I don't think it was helpful at all.

http://wiki.znc.in/WeeChat is also awaiting the capability and the safe capabilities.

**Comment by maxteufel** [comment]
> Is there something wrong with PR or when can we expect this to come to WeeChat?

I don't plan to change anything in this PR, I tested it and it works.

> If I have cap-notify enabled and away-notify in irc.server.blah.capabilities and blah starts offering away-notify, will it be automatically enabled?

Yes, all capabilities enabled in the config should be requested when the server starts offering it (note this doesn't support SASL reauth).

File: src/plugins/irc/irc-protocol.c [hunk 1]
```diff
@@ -514,6 +514,98 @@ IRC_PROTOCOL_CALLBACK(cap)
...
+        {
+            ptr_caps = (argv_eol[4][0] == ':') ? argv_eol[4] + 1 : argv_eol[4];
+            weechat_printf_date_tags (
+                server->buffer, date, NULL,
+                _("%s%s: client capability, now available: %s"),
+                weechat_prefix ("network"), IRC_PLUGIN_NAME, ptr_caps);
+
+            /* Assume that we're not requesting any already-enabled capabilities
+             * TODO SASL Reauthentication */
+            ptr_cap_option = IRC_SERVER_OPTION_STRING(server,
```
Comment (talisein) [inline src/plugins/irc/irc-protocol.c]: This is all nearly copy-paste from 366-408. Why not make a cap parsing function?

**Comment by Mikaela** [comment]
At IRC it was said that it helps if we merge these manually and test. On the latest WeeChat from just now, this appears to work:

```
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: account-notify cap-notify extended-join multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: account-notify away-notify cap-notify extended-join multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: cap-notify multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: cap-notify multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: cap-notify multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: cap-notify multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: account-notify away-notify cap-notify extended-join multi-prefix server-time userhost-in-names
2015-10-18 10:37:18+0300 -- irc: client capability, currently enabled: multi-prefix userhost-in-names
```

As I use ZNC, at least away-notify and account-notify wouldn't appear otherwise.

**Pull request merged**